### DISCUSSION TITLE
Enable live code editing for something other than JS?

### DISCUSSION THREAD

**Discussion by depombo** [facebook/docusaurus#5170] [Q&A]
Hi all, I'm trying to enable live code editing for a programming language that is not JS, but can be transpiled to JS. This involves an extra compilation step before the live codeblock theme's internal one. Ideally I can wrap/swizzle a theme's component (It is not clear which exact component it should be though) when the block is of this specific language?

**Comment by slorber** [comment]
I think you can swizzle `@theme/Playground` of docusaurus-theme-live-codeblock, and use the `transformCode` prop here with your own logic

```js
      <LiveProvider
        key={isClient}
        code={isClient ? children.replace(/\n$/, '') : ''}
        transformCode={transformCode || ((code) => `${code};`)}
        theme={prismTheme}
        {...props}>
        {playgroundPosition === 'top' ? (
          <>
            <ResultWithHeader />
            <EditorWithHeader />
          </>
        ) : (
          <>
            <EditorWithHeader />
            <ResultWithHeader />
          </>
        )}
      </LiveProvider>
```


> This involves an extra compilation step before the live codeblock theme's internal one.

Remember that the code transformation must be runnable in the browser (synchronously), you can't use a nodejs package here



---

> (It is not clear which exact component it should be though)

Swizzling means you opt-in for current implementation details. So you generally have to understand those implementation details in the first place and figure out yourself which components to "override" because in the end you'll have to maintain those overrides yourself as Docusaurus updates them. Please make sure to read the source code fo this theme and to understand it before even thinking about overriding it, because we can't provide support for that and you are now on your own.

> when the block is of this specific language

It may not be necessary, I believe the playground will be live if you use the `live` codeblock param, no matter the language: ` ```clojure live ` should work

**Reply by depombo** [reply]
> I think you can swizzle `@theme/Playground` of docusaurus-theme-live-codeblock, and use the `transformCode` prop here with your own logic
> 
> ```js
>       <LiveProvider
>         key={isClient}
>         code={isClient ? children.replace(/\n$/, '') : ''}
>         transformCode={transformCode || ((code) => `${code};`)}
>         theme={prismTheme}
>         {...props}>
>         {playgroundPosition === 'top' ? (
>           <>
>             <ResultWithHeader />
>             <EditorWithHeader />
>           </>
>         ) : (
>           <>
>             <EditorWithHeader />
>             <ResultWithHeader />
>           </>
>         )}
>       </LiveProvider>
> ```
> 
>  > This involves an extra compilation step before the live codeblock theme's internal one.
> 
> Remember that the code transformation must be runnable in the browser (synchronously), you can't use a nodejs package here
> 
> > (It is not clear which exact component it should be though)
> 
> Swizzling means you opt-in for current implementation details. So you generally have to understand those implementation details in the first place and figure out yourself which components to "override" because in the end you'll have to maintain those overrides yourself as Docusaurus updates them. Please make sure to read the source code fo this theme and to understand it before even thinking about overriding it, because we can't provide support for that and you are now on your own.
> 

This is helpful, thank you! 

> > when the block is of this specific language
> 
> It may not be necessary, I believe the playground will be live if you use the `live` codeblock param, no matter the language: `` ```clojure live `` should work

It seems like the syntax highlighting for all live code blocks are JS so I will have to dive into the internals to figure out how to override that
